app/models/spree/gateway/pay_pal_express.rb

```diff
@@ -33,6 +33,62 @@ def auto_capture?
     def method_type
       'paypal'
     end
+    
+    # This is rather hackish, required for payment/processing handle_response code.
+    def empty_success
+      Class.new do
+        def success?; true; end
+        def authorization; nil; end
+      end.new
+    end
+
+    #this is essentially the refund method below except the creation of the refund transaction
+    #as that will count doubly towards the "credited" amount
+    def void(response_code, gateway_options={})
+      payment = Spree::Payment.find_by_response_code(response_code)
```

Comment:
Finding a Payment by response code is problematic, as I've found out that all PayPal Payments have a `transaction_id` set, but `response_code` is always `nil`. The `response_code` in the parameter is also always `nil` here, which leads to a very serious problem: The only `Payment` ever found will be the very first one without a `response_code`, which can also be a `CreditCard`-Payment or anything else. Unfortunately, I have not yet found a proper fix.